improver/calibration/ensemble_calibration.py

```diff
@@ -1115,35 +1124,37 @@ def guess_and_minimise(
 
         """
         if self.each_point:
-            # Disable ordinary least squares fit when computing each point
-            # independently due to potential for too few data points to create
-            # a reliable fit.
-            if self.predictor == "realizations":
-                self.ESTIMATE_COEFFICIENTS_FROM_LINEAR_MODEL_FLAG = False
             index = [
                 forecast_predictor.coord(axis="y"),
                 forecast_predictor.coord(axis="x"),
             ]
-
             argument_list = (
                 (
                     truths_slice.data,
                     fp_slice.data,
                     self.predictor,
-                    self.ESTIMATE_COEFFICIENTS_FROM_LINEAR_MODEL_FLAG,
+                    False
+                    if np.any(np.isnan(truths_slice.data))
+                    or np.any(np.isnan(fp_slice.data))
+                    else self.ESTIMATE_COEFFICIENTS_FROM_LINEAR_MODEL_FLAG,
```

Comment:
I think that this is essentially working as you suggest. Perhaps it's useful to point out that this check is within a for loop over the truth and forecast points. I think that this is working as:

1. Loop over all sites. 
2. Check whether there are any NaNs within the training dataset for that particular site. 
3. If there are NaNs then use the default initial guess for that particular site, otherwise compute the initial guess.

I think that this is addressing your concern that it's best to to compute an initial guess for points, rather than using the default, as long as NaNs aren't present. Hopefully that's correct.

Reply:
That is what I wanted, so it sounds like its doing it. Thanks.